liblouis/compileTranslationTable.c

```diff
@@ -2838,143 +2834,154 @@ compileRule(FileInfo *nested, TranslationTableHeader **table,
+				else
+					len = _lou_pattern_compile(&ptn_before.chars[0], ptn_before.length,
+							&patterns[1], 13841, *table, nested);
+				if (len) {
+					mrk = patterns[0] = len + 1;
+					_lou_pattern_reverse(&patterns[1]);
+
+					if (ptn_after.chars[0] == '-' && ptn_after.length == 1)
+						len = _lou_pattern_compile(&ptn_after.chars[0], 0, &patterns[mrk],
+								13841, *table, nested);
+					else
+						len = _lou_pattern_compile(&ptn_after.chars[0], ptn_after.length,
+								&patterns[mrk], 13841, *table, nested);
+					if (len) {
+						len += mrk;
+
+						if (allocateSpaceInTranslationTable(nested, &patternOffset,
+									len * sizeof(widechar), table)) {
+							ok = 1;
+
+							/* realloc may have moved table, so make sure rule is still
+							 * valid */
+							rule = (TranslationTableRule *)&(*table)
+										   ->ruleArea[ruleOffset];
+
+							memcpy(&(*table)->ruleArea[patternOffset], patterns,
+									len * sizeof(widechar));
+							rule->patterns = patternOffset;
+						}
+					}
+				}
 			}
 
-			/* realloc may have moved table, so make sure rule is still valid */
-			rule = (TranslationTableRule *)&(*table)->ruleArea[ruleOffset];
-
-			memcpy(&(*table)->ruleArea[patternOffset], patterns, len * sizeof(widechar));
-			rule->patterns = patternOffset;
-
-			break;
+			free(patterns);
+			return ok;
 		}
 
 		case CTO_BegCapsPhrase: {
 			// not passing pointer because compileBrailleIndicator may reallocate table
 			TranslationTableOffset ruleOffset =
 					(*table)->emphRules[capsRule][begPhraseOffset];
-			ok = compileBrailleIndicator(nested, "first word capital sign",
-					CTO_BegCapsPhraseRule, &ruleOffset, noback, nofor, table);
-			(*table)->emphRules[capsRule][begPhraseOffset] = ruleOffset;
-			break;
+			if (compileBrailleIndicator(nested, "first word capital sign",
+						CTO_BegCapsPhraseRule, &ruleOffset, noback, nofor, table)) {
+				(*table)->emphRules[capsRule][begPhraseOffset] = ruleOffset;
+				return 1;
+			}
+			return 0;
 		}
 		case CTO_EndCapsPhrase: {
 			TranslationTableOffset ruleOffset;
 			switch (compileBeforeAfter(nested)) {
 			case 1:  // before
 				if ((*table)->emphRules[capsRule][endPhraseAfterOffset]) {
 					compileError(nested, "Capital sign after last word already defined.");
-					ok = 0;
-					break;
+					return 0;
 				}
 				// not passing pointer because compileBrailleIndicator may reallocate
 				// table
 				ruleOffset = (*table)->emphRules[capsRule][endPhraseBeforeOffset];
-				ok = compileBrailleIndicator(nested, "capital sign before last word",
-						CTO_EndCapsPhraseBeforeRule, &ruleOffset, noback, nofor, table);
-				(*table)->emphRules[capsRule][endPhraseBeforeOffset] = ruleOffset;
-				break;
+				if (compileBrailleIndicator(nested, "capital sign before last word",
+							CTO_EndCapsPhraseBeforeRule, &ruleOffset, noback, nofor,
+							table)) {
+					(*table)->emphRules[capsRule][endPhraseBeforeOffset] = ruleOffset;
+					return 1;
+				}
```

Comment:
`else` would be more explicit here (and in some other places). Personal preference I guess.

Reply:
What if was be rewritten to the early return style that you said you liked better?

```
if (!compileBrailleIndicator(...))
    return 0;
(*table)->emphRules[capsRule][endPhraseBeforeOffset] = ruleOffset;
return 1;
```

This also does not have an `else`? Are your preferences consistent? 😄 

I think the reason why I like putting a return at the end of a `case` section is for my peace of mind: to make sure the control flow can never continue beyond that point.

Reply:
Yes, I'd prefer early return style.